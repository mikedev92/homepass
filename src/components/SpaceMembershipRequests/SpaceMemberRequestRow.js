// @flow
import React from 'react';
import { Right, Button, ListItem, Left, Body, Thumbnail, Text, Row } from 'native-base';
import moment from 'moment';
import { variables } from '../../theme';

type Props = {
  spaceMemberRequest?: SpaceMemberRequest | null,
  last: boolean,
  approveSpaceMemberRequest: string => () => void,
  rejectSpaceMemberRequest: string => () => void,
};

function SpaceMemberRequestRow({
  spaceMemberRequest,
  last,
  approveSpaceMemberRequest,
  rejectSpaceMemberRequest,
}: Props) {
  if (!spaceMemberRequest || !spaceMemberRequest.user) {
    return null;
  }

  const spaceMemberRequestId = spaceMemberRequest.spaceMemberRequestId;
  if (typeof spaceMemberRequestId !== 'string') {
    return null;
  }

  return (
    <ListItem key={spaceMemberRequest.spaceMemberRequestId} thumbnail last={last}>
      {spaceMemberRequest.user.imageUrl &&
        <Left><Thumbnail source={{ uri: spaceMemberRequest.user.imageUrl }} /></Left>}
      <Body>
        <Text>{spaceMemberRequest.user.name}</Text>
        <Text note>{moment(spaceMemberRequest.createdAt).fromNow()}</Text>
        <Row style={{ marginTop: variables.contentPadding }}>
          <Button
            onPress={approveSpaceMemberRequest(spaceMemberRequestId)}
            style={{ marginRight: variables.contentPadding }}
            small
          >
            <Text>Approve</Text>
          </Button>
          <Button onPress={rejectSpaceMemberRequest(spaceMemberRequestId)} small bordered>
            <Text>Decline</Text>
          </Button>
        </Row>
      </Body>
      <Right />
    </ListItem>
  );
}

export default SpaceMemberRequestRow;
