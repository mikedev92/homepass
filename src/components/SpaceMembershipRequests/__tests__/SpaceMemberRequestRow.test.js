// @flow
import React from 'react';
import { shallow } from 'enzyme';
import MockDate from 'mockdate';
import SpaceMemberRequestRow from '../SpaceMemberRequestRow';

const testProps = [
  {
    approveSpaceMemberRequest: jest.fn(),
    rejectSpaceMemberRequest: jest.fn(),
    spaceMemberRequest: {
      id: '123',
      spaceMemberRequestId: '123',
      createdAt: new Date(1234).toISOString(),
      user: null,
    },
    last: true,
  },
  {
    approveSpaceMemberRequest: jest.fn(),
    rejectSpaceMemberRequest: jest.fn(),
    spaceMemberRequest: {
      id: '123',
      spaceMemberRequestId: '123',
      createdAt: new Date(1234).toISOString(),
      user: {
        id: '123',
        userId: '123',
        name: 'Name',
        firstName: 'first name',
        lastName: 'last name',
        imageUrl: 'Image',
        email: null,
        emailVerified: false,
        lastLoginAt: new Date(1234).toISOString(),
        layerId: null,
        mobile: null,
        mobileVerified: false,
      },
    },
    last: true,
  },
];
describe('<SpaceMemberRequestRow />', () => {
  beforeEach(() => {
    MockDate.set(new Date('1/1/2000'));
  });

  afterEach(() => {
    MockDate.reset();
  });
  it('matches snapshots', () => {
    testProps.forEach((props) => {
      const wrapper = shallow(<SpaceMemberRequestRow {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
