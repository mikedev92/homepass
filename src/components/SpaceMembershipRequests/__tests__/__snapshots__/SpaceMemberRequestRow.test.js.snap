// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpaceMemberRequestRow /> matches snapshots 1`] = `null`;

exports[`<SpaceMemberRequestRow /> matches snapshots 2`] = `
<Styled(ListItem)
  last={true}
  thumbnail={true}
>
  <Styled(Left)>
    <Styled(Thumbnail)
      source={
        Object {
          "uri": "Image",
        }
      }
    />
  </Styled(Left)>
  <Styled(Body)>
    <Styled(Text)>
      Name
    </Styled(Text)>
    <Styled(Text)
      note={true}
    >
      30 years ago
    </Styled(Text)>
    <RowNB
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <Styled(Button)
        small={true}
        style={
          Object {
            "marginRight": 10,
          }
        }
      >
        <Styled(Text)>
          Approve
        </Styled(Text)>
      </Styled(Button)>
      <Styled(Button)
        bordered={true}
        small={true}
      >
        <Styled(Text)>
          Decline
        </Styled(Text)>
      </Styled(Button)>
    </RowNB>
  </Styled(Body)>
  <Styled(Right) />
</Styled(ListItem)>
`;
